blueprints/addon/index.js

```diff
@@ -57,6 +57,10 @@ module.exports = {
     contents.dependencies['ember-cli-htmlbars'] = contents.devDependencies['ember-cli-htmlbars'];
     delete contents.devDependencies['ember-cli-htmlbars'];
 
+    // Move ember-auto-import into the dependencies of the addon blueprint by default
+    contents.dependencies['ember-auto-import'] = contents.devDependencies['ember-auto-import'];
```

Comment:
We should discuss this a bit more. I'm not sure that we want to make all addons depend on ember-auto-import 🤔 

@Turbo87 / @ef4 - Thoughts?

Reply:
yeah, that's what I said on Discord too. I think it's fine adding it to apps, but I think for addons the authors should have to manually add it if they really need to. I'm assuming that with embroider the whole story will change again anyway.

Reply:
Yeah, I wasn't sure but it *was* included in the octane blueprints so I threw it in there. Also brought it in the #st-octane channel in Discord and @chancancode said "it's a bug, it should include it", linking to this: https://www.notion.so/Octane-Preview-3-13-FAQ-and-Known-Issues-bbd25c2ffb0e4f079103db1fc0144175#284a1ee8f26448d6bd77febc2583aa7f

Reply:
I hate to be a bummer, but it would be nice to discuss if it is ready to be added by default at all. More here https://github.com/ember-cli/ember-cli/issues/8872#issuecomment-538023305.

Reply:
I agree, I think adding ember-auto-import to the default blueprint has significant ecosystem-wide effects that need to be aired in an RFC. My own notes on what I see as the desirable TODOs before doing that are here: https://github.com/ef4/ember-auto-import/issues/182

Reply:
I want to clarify that my caveats are almost all about usage by *addons*. Having it in apps is less risky and a huge ergonomics win, so I'm in favor of having it in the app blueprint.

(The difference is: addons need to just work with whatever version of auto-import their consuming app is using. Even future ones. We can gracefully roll out breaking changes to app authors, we cannot easily to addon authors.)

Reply:
Sounds like there is a relative consensus to include it for apps but not addons.?. Let me know which way it needs to go and I'll make the changes, or feel free to push a commit to the PR.

Reply:
Ya, I think we are on the same page now. Apps should get it by default, and addons would only have it for their dummy apps (in other words, it should stay in `devDependencies`).

Reply:
@kellyselden - I did some local tests between a 3.14.0-beta.2 app with and without `ember-auto-import` as a dependency. Adding `ember-auto-import` adds _roughly_ 2s to the initial build time (for cold builds they went from 17.67s to 19.73s; warm builds went from 7.94 to 8.96), and rebuild time approximately 93ms (went from 213ms to 306ms). These times seem fine to me (considering the significant improvement to development ergonomics).